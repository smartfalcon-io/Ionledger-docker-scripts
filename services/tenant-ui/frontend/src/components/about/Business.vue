<template>
  <div class="grid">
    <div class="col-3">{{ config.frontend.ux.aboutBusiness.linkTitle }}</div>
    <div class="col-9 md:col-6">
      <a :href="config.frontend.ux.aboutBusiness.link">{{
        config.frontend.ux.aboutBusiness.link
      }}</a>
    </div>
    <div
      v-if="config.frontend.ux.aboutBusiness.imageUrl"
      class="col-12 md:col-3 flex justify-content-end pt-0"
    >
      <img
        :src="config.frontend.ux.aboutBusiness.imageUrl"
        class="logo-business"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { storeToRefs } from 'pinia';
import { useConfigStore } from '@/store/configStore';
const { config } = storeToRefs(useConfigStore());
</script>

<style scoped>
.logo-business {
  width: 18em;
}
</style>
