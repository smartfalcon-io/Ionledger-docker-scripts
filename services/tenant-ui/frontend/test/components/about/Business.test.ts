import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import PrimeVue from 'primevue/config';
import { describe, expect, test } from 'vitest';

import Business from '@/components/about/Business.vue';

describe('Business', async () => {
  test('mount matches snapshot with expected values', () => {
    const wrapper = mount(Business, {
      global: {
        plugins: [PrimeVue, createTestingPinia()],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
