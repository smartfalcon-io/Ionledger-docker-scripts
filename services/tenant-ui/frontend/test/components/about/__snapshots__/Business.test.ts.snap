// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Business > mount matches snapshot with expected values 1`] = `
"<div data-v-0eee3199="" class="grid">
  <div data-v-0eee3199="" class="col-3">About Business</div>
  <div data-v-0eee3199="" class="col-9 md:col-6"><a data-v-0eee3199="" href="http://link.com">http://link.com</a></div>
  <div data-v-0eee3199="" class="col-12 md:col-3 flex justify-content-end pt-0"><img data-v-0eee3199="" src="http://image.com" class="logo-business"></div>
</div>"
`;
